btree/btree.c

```diff
@@ -622,6 +625,71 @@ enum {
 	INTERNAL_NODE_VALUE_SIZE = sizeof(void *),
 };
 
+#define M0_BTREE_TX_CAPTURE(tx, seg, ptr, size)                              \
+			   m0_be_tx_capture(tx, &M0_BE_REG(seg, size, ptr))
+
+#if AVOID_BE_SEGMENT
+
+#undef M0_BTREE_TX_CAPTURE
+#define M0_BTREE_TX_CAPTURE(tx, seg, ptr, size)                              \
+	({                                                                   \
+		typeof(size) __size = (size);                                \
+		(tx) = (tx);                                                 \
+		(seg) = (seg);                                               \
+		(ptr) = (ptr);                                               \
+		(__size) = (__size);                                         \
+	})
```

Comment:
It will be better to have it in do while(0)

Reply:
Will do.